src/gpodder/feedcore.py

```diff
@@ -171,10 +173,13 @@ def _parse_feed(self, url, etag, modified, autodiscovery=True):
             except HTTPError as e:
                 return self._check_statuscode(e, e.geturl())
 
+        data = stream
         if not is_local and stream.headers.get('content-type', '').startswith('text/html'):
             if autodiscovery:
```

Comment:
Since this change removes the other branch (the thrown exception), this conditional could be combined with the one above.